File: banda/IvanRC-banda-Java/src/main/java/modelsBandaMusicos/Bajista.java
package modelsBandaMusicos;

public class Bajista extends Musico implements IBajista{
    String nombre;
    Double salario = 1500.00;
    int añosExperiencia;
    int numeroCuerdas;

    public Bajista(String nombre, int añosExperiencia, int numeroCuerdas){
        super(nombre, añosExperiencia);
        this.numeroCuerdas = numeroCuerdas;
    }

    public double recalcularSalario(double salario) {
        return IBajista.super.recalcularSalario(salario);
    }

    public int getNumeroCuerdas() {
        return 0;
    }
}

File: banda/SergioSF-Conservatorio-Kotlin/src/main/kotlin/models/Bajista.kt
package models

class Bajista(nombre: String, apellidos: String, edad: Int, anosExperiencia: Int) : Musico(
    nombre, apellidos, edad,
    anosExperiencia
), IBajista {
    override var instrumento: Instrumento = Instrumento.BAJO
    override var especialidad: Especialidad = Especialidad.BAJISTA

    override var salario: Int = (super.salario * 1.30).toInt()
    override var representacion: Int = 10

    override fun interpretar() {
        tocarBajo()
    }

    override fun presentarse() {
        println("Hola soy $nombre, $apellidos, tengo $edad años y llevo $anosExperiencia años siendo músico. Mi instrumento ${Instrumento.TROMPETA}")
    }

    override fun tocarBajo() {
        println("*toca el bajo")
    }

    override fun toString(): String {
        return println("$salario").toString()
    }
}
